drivers/clock_control/clock_control_tisci.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2025, Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_k2g_sci_clk
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/firmware/tisci/tisci.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/tisci_clock_control.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ti_k2g_sci_clk, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+const struct device *dmsc = DEVICE_DT_GET(DT_NODELABEL(dmsc));
+
+static int tisci_get_rate(const struct device *dev, clock_control_subsys_t sys, uint32_t *rate)
+{
+	struct tisci_clock_config *req = (struct tisci_clock_config *)sys;
+	uint64_t temp_rate;
+	int ret = tisci_cmd_clk_get_freq(dmsc, req->dev_id, req->clk_id, &temp_rate);
```

Comment:
Declare `ret`, then a newline, then assign to it, then no newline after between the assignment and the checking of `ret`. Same for ` tisci_set_rate()`, make them look like you did in `tisci_get_status()`.